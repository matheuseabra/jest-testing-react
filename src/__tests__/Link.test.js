import React from 'react';
import renderer from 'react-test-renderer';
import Link from '../components/Link';

it('Link component changes className when hovered', () => {
    const linkComponent = renderer.create(<Link page="https://www.twitter.com">Twitter</Link>);
    let linkTree = linkComponent.toJSON();
    expect(linkTree).toMatchSnapshot();
  
    linkTree.props.onMouseEnter();
    linkTree = linkComponent.toJSON();
  
    linkTree.props.onMouseLeave();
    linkTree = linkComponent.toJSON();
  
    expect(linkTree).toMatchSnapshot();
  });