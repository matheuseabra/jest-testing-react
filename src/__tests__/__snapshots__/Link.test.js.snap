// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component changes className when hovered 1`] = `
<a
  className="normal"
  href="https://www.twitter.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Twitter
</a>
`;

exports[`Link component changes className when hovered 2`] = `
<a
  className="hovered"
  href="https://www.twitter.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Twitter
</a>
`;
